core/src/dsnp/pseudo_relationship_identifier.rs

```diff
@@ -0,0 +1,79 @@
+use crate::dsnp::dsnp_types::DsnpPrid;
+use anyhow::{Error, Result};
+use crypto_box::{aead::AeadInPlace, Nonce, PublicKey, SecretKey};
+use hkdf::Hkdf;
+use sha2::Sha256;
+use std::ops::Deref;
+use x25519_dalek::StaticSecret;
+use xsalsa20poly1305::{KeyInit, XSalsa20Poly1305};
+use zeroize::Zeroizing;
+
+const PRI_CONTEXT: &[u8] = b"PRIdCtx0";
+struct PseudonymousRelationshipIdentifier;
+
+impl PseudonymousRelationshipIdentifier {
+	fn create(
+		a: u64,
+		b: u64,
+		a_secret_key: &SecretKey,
+		b_public_key: &PublicKey,
+	) -> Result<DsnpPrid> {
+		let id_b = b.to_le_bytes();
+
+		// calculate shared secret
+		let root_shared = StaticSecret::from(*a_secret_key.as_bytes())
+			.diffie_hellman(&x25519_dalek::PublicKey::from(*b_public_key.as_bytes()));
+
+		// derive a new key form pri context
+		let mut derived_key = Zeroizing::new([0u8; 32]);
+		let hk = Hkdf::<Sha256>::new(Some(&id_b[..]), root_shared.as_bytes());
```

Comment:
Unfortunately `Hkdf` does not support `Blake2` for hashing! so had to use `Sha256`

Reply:
Hmm, that's kind of a problem for compatibility purposes. Is there a workaround?

Reply:
Alternatively, we could manually do the libsodium key derivation, which they specify as

```
BLAKE2B-subkeylen(key=key, message={}, salt=subkey_id || {0}, personal=ctx || {0})
```

...probably good to have a test vector to verify against, of course.

(https://doc.libsodium.org/key_derivation)

Reply:
Found a native rust libsodium implementation that seems like it is completely compatible with the C version. So far compatibility testing works as expected
https://crates.io/crates/dryoc